internals/overlord/state/identities.go

```diff
@@ -264,6 +264,22 @@ func (s *State) Identities() map[string]*Identity {
 	return result
 }
 
+// IdentityFromInputs returns an identity with the given inputs, or nil
+// if there is none.
+func (s *State) IdentityFromInputs(userID *uint32) *Identity {
```

Comment:
Maybe add to the comment that this requires the state lock to be held, also for as long as the returned `*Identity` is accessed? Or is `s.reading()` taking care of checking this?

Reply:
Most of these state functions assume but don't explicitly document that you need to hold the state lock, so I think this is okay as is (and yeah, the `s.reading()` checks).